### ISSUE TITLE
lodash/fp document like the one at https://lodash.com/docs

### ISSUE THREAD

**Issue by enricofadly** [lodash/lodash#4890]
im new in fp js programming and i see that the lodash/fp is a good tool since its automatically curried and data last parameter, but the problem is i couldnt find proper documentation on this like the one at lodash non-fp at the site, 

can i request this doc feature or maybe u guys could point me to resource who already docemented this lodash/fp nice?

Thanks in advance.

**Comment by LarryBattle** [comment]
- https://github.com/lodash/lodash/wiki/FP-Guide
- https://simonsmith.io/dipping-a-toe-into-functional-js-with-lodash-fp
- https://www.youtube.com/watch?v=Tx-0tcEl1I0

**Comment by bnjmnt4n** [comment]
Thanks for the links @LarryBattle! We don't have additional documentation for `lodash/fp` since it's largely based on our plain lodash builds.

**Comment by baer** [comment]
Full-fledged FP documentation is something I've been interested in for a long time. Lodash is an unbelievable gem in the JavaScript ecosystem and, in particular, the documentation is (and always has been) among the best I've ever used. It's the benchmark I use to judge other projects so I've always wondered why FP was just a Markdown wiki. 

Today, I decided to find out. Below is (hopefully) a somewhat complete history of FP documentation. If I've misrepresented anybody involved, it was not my intention. Please help me correct any errors or omissions if you see them! Github's search is all I had to go on 😃.

## Once Upon a Time...
In early 2016, when ES2015 was still a 👶, @jfmengels submitted a pair of PRs "[generate docs for lodash/fp ... [for] the documentation page of the website](https://github.com/lodash/lodash/pull/1897)". The response from @jdalton at the time was, "[Let's do it!](https://github.com/lodash/lodash/pull/1897#issuecomment-176526237)." See #1897 #1963. Both PRs were quickly merged into the main codebase. After about a week, @jdalton [decided to take things in a different direction](https://github.com/lodash/lodash/commit/cf1a4f893ff3e72a2fb4cab48a613377beb467b6#r16182602) and @jfmengels broke the FP-documentation project into a separate repo called [lodash-fp-docs](https://github.com/jfmengels/lodash-fp-docs/commit/3bbeb405877812a01049a925ccb92d94f19ec4f8). At that time [@bnjmnt4n seemed hopeful](https://github.com/lodash/lodash/issues/1914) that this work would lead to new FP docs. As late as August @jdalton indicated that lodash-fp-docs may turn into something more official [here](https://github.com/lodash/lodash/issues/2566#issuecomment-238272644) and [here](https://github.com/lodash/lodash/issues/1932#issuecomment-178992694). Fast forward a bit and the message from lodash is a bit more tempered with @bnjmnt4n's comment in this PR as well as @jdalton's 2017 comment that "[I think the level of docs is appropriate](https://github.com/lodash/lodash/issues/3173#issuecomment-304318259),"  and [a 2018 reference to lodash-fp-docs](https://github.com/lodash/lodash/issues/3717#issuecomment-376448014). 

Reading through the issues related to lodash/fp, it seems like there may be a surprising bit of complexity maintaining these docs. In addition to the function signatures, some of the example code needs to change to reflect lodash/fp's lack of mutation. 

## One More Thing...
Before I consider this issue closed, I have to ask: would you be interested in a PR that generated the lodash/fp docs without examples? The docs are just so stupendous! :)

## Issues involving lodash/fp docs

#1897
#1932
#1963
#2566
#2804
#3173
#3717

**Comment by jfmengels** [comment]
@baer Thanks for the history summary :slightly_smiling_face: 

I have just updated the docs in my public Gist to 4.17.20 https://gist.github.com/jfmengels/6b973b69c491375117dc


> One More Thing...
> Before I consider this issue closed, I have to ask: would you be interested in a PR that generated the lodash/fp docs without examples? The docs are just so stupendous! :)

Just in case it isn't clear to all, the docs I linked to above are automatically generated from the Lodash.js and fp mapping source code files, using automatic re-ordering of arguments in the docs and in the examples.

I think that the Gist is already quite goodn and it is not necessarily a problem of maintenance, because these docs are not necessarily hard to maintain (we could automate the release pretty easily from where we are). It's just that some examples don't make sense when used in an FP scenario, and others don't show-case the capabilities of the FP scenario.

Instead of removing all examples, I think it would be nicer to have a special case in the code generation code for each example that needs to be improved, where instead of taking the example from the source code, we inject a better one. Once that is down, I think there will be quality documentation for lodash/fp.

Back when I was working on this, I was playing/learning with ASTs and code generation (which has now served me well, I built a [linter for a different programming language](https://package.elm-lang.org/packages/jfmengels/elm-review/latest/)), but there were obvious cases where the examples could never be adapted automatically.

I stopped working on this back in the day because the docs served me well enough and I didn't want to hardcode/deviate from the examples from the main project. But I now believe that that is an acceptable solution to get the docs :ok_hand:
